Include/cpython/object.h

```diff
@@ -256,6 +256,9 @@ typedef struct _typeobject {
     destructor tp_finalize;
     vectorcallfunc tp_vectorcall;
 
+    /* Unused, kept for backwards compatibility in CPython 3.8 only */
+    int (*tp_print)(PyObject *, FILE *, int);
```

Comment:
Does adding the `Py_DEPRECATED(3.8)` macro here cause the `tp_print` reference in the test to generate a compiler warning?

If it does, that would be a good thing to add (as long as we also have a way to disable the deprecation warning in the test suite, which I presume we do)

Reply:
That works but then the deprecation warning will be shown when compiling `_testcapi.c`. I'll add a guard for that. If that's controversial, I'll gladly remove it.

(off-topic: since this is the only such warning shown, it seems that we never actually test deprecated functionality of the C API, which is worrying)